assets/javascripts/discourse/components/result-types/json.js

```diff
@@ -0,0 +1,35 @@
+import Component from "@glimmer/component";
+import FullscreenCodeModal from "discourse/components/modal/fullscreen-code";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+import { cached } from "@glimmer/tracking";
+
+export default class Json extends Component {
+  @service dialog;
+  @service modal;
+
+  @cached
+  get parsedJson() {
+    try {
+      return JSON.parse(this.args.ctx.value);
+    } catch {
+      return null;
+    }
+  }
+
+  get elidedValue() {
+    return this.args.ctx.value.slice(0, 50) + "...";
+  }
+
+  @action
+  viewJson() {
+    this.modal.show(FullscreenCodeModal, {
```

Comment:
We should avoid doing this, it's considering any modal we write in core as public API.

Not sure what platform would prefer here instead, worth discussing it.

Reply:
Fair enough, but AFAIK we don't have any patterns for this yet and it would be nice to not hold this up until TGX is back from holiday. This modal isn't likely to change anytime soon.

Reply:
I will make a topic about it too.